### ISSUE TITLE
Search Experience Redesign - Feedback on Filters and Tag Clouds

### ISSUE THREAD

**Issue by SudharakaP** [opencollective/opencollective#5327]
This is related the search experience redesign project; https://github.com/opencollective/opencollective/issues/4295

I am opening this discussion issue as I felt we need to clarify which filters we need exactly and what kinda tags should be shown by default. 

The current design draft for this is at, 

https://www.figma.com/file/KinuveYf4WUlRIXzMTnmIsmO/%5BOC-Design%5D-Lab?node-id=14314%3A189472

Here is the summary of ideas which discussed in the kickoff (and found in comments in the [project issue](https://github.com/opencollective/opencollective/issues/4295)); 

- Add a country filter 
- Add a currency filter
- What should we have for the tag cloud? Currently in the discover page we have the following tags; `Open Source Collectives`, `COVID-19 Collectives`, `Pledged Collectives` and `Other`. Given that we've decided to remove `Pledged Collectives` we are left with two tags. Also note that when we create a collective (using the `For any Community`) option we add a `Community` tag by default. Do we need to add this default tag as well to the list? 
- Sorting: Currently in the discover page we have `Most Popular` and `Newest` sort options. In the [project issue it was suggested](https://github.com/opencollective/opencollective/issues/4295#issuecomment-1062040889) to have `Creation Date` and `Activity`. 

Feel free to comment on this thread on any ideas you have related to the filters an tag clouds. My goal is to come to a consensus on what we need specifically so that we could assess the possibility of supporting them, align the designs accordingly and break them into smaller tasks for the next couple of weeks for implementation.

cc: @znarf @kaylarep @Memo-Es @BenJam @piamancini @alanna

**Comment by Memo-Es** [comment]
**Regarding the tag cloud:**
* Let's start with fetching some data on what the current tags are for existing collectives, we can start displaying the tags most used
* Going forward we can include here suggestions we want new Collectives to use (e.g., trending like 'ukraine-aid' or something like that)

**Regarding the location filter**
* Still not sure about how we are nailing this one down if it's correct to leave 'Country' plus a 'Region' or have a custom 'Location' one with options like: US, Europe, UK, NZ, etc.

**Comment by kaylarep** [comment]
> **Regarding the tag cloud:**
> 
> * Let's start with fetching some data on what the current tags are for existing collectives, we can start displaying the tags most used
> * Going forward we can include here suggestions we want new Collectives to use (e.g., trending like 'ukraine-aid' or something like that)

+1 to this.  The tag clouds I'm familiar with are sort of auto-created, with the top-used tags either appearing the biggest or at the top.


> **Regarding the location filter**
> 
> * Still not sure about how we are nailing this one down if it's correct to leave 'Country' plus a 'Region' or have a custom 'Location' one with options like: US, Europe, UK, NZ, etc.

If it's possible, I would prefer to have a 'Country' drop-down. And then once the country is selected another box appears that says either 'State' or 'Providence' or whatever that country uses. If the country doesn't need to be broken down in this way, then no second box would appear.

Also perhaps the option to select multiple countries could be good (because some 'Regions' are multiple countries).

Also, perhaps in the 'Country' drop down, the first option could be 'virtual' or 'remote' for groups that don't have a centralized location (or they just wouldn't select any country and keep at 'All')


--Please let me know if this needs any more clarification, what you think.

**Comment by alanna** [comment]
@kaylarep can you share the tags list you've been working on for OCF Collectives with Memo and Suds? I think that will help them understand the types of tags we think will be useful.

> Currently in the discover page we have the following tags; Open Source Collectives, COVID-19 Collectives, Pledged Collectives and Other

I think we should ditch this concept entirely and just show a tag cloud with the most used tags first. Any category we select will soon go out of date.

> Also note that when we create a collective (using the For any Community) option we add a Community tag by default. Do we need to add this default tag as well to the list?

We should remove this. Community is too generic and doesn't really mean anything. Ideally we should implement #4764 with a more nuanced design, but until then, let's take the concept of adding tags out of on-boarding to simplify.

Sometimes there is a strong case for adding tags based on host, but this should be quite limited as we don't want to add tags incorrectly. For example, we don't want to add `US` to all OCF Collectives, even though it's true 99%, because there are a few exceptions. But OSC Collectives can confidently get the `open source` tag, and AllForClime ones can get the `climate` tag. In the future I want to discuss Hosts being able to add tags to their Collectives or being able to set tags automatically applied to all their Collectives as they join. All this out of scope for this iteration though. For now, let's take tags out of onbaording.

> Sorting: Currently in the discover page we have Most Popular and Newest sort options. In the https://github.com/opencollective/opencollective/issues/4295#issuecomment-1062040889 to have Creation Date and Activity.

Creation date and activity are good... can we also do something with amount of contributions?

How will we deal with all the spam profiles? There are many Collectives people have created as demo/test and added huge amounts to. if someone sorts by most raised, they will get a bunch of spam and have a bad experience.

> Still not sure about how we are nailing this one down if it's correct to leave 'Country' plus a 'Region' or have a custom 'Location' 

In all cases where we capture this data another way (such as the country setting that exists separately from tags) we should enable search filters to use that info effectively.

The issue is different areas have different needs for location filtering. In Europe, sometimes `EU` is the important part, sometimes it's `country` and sometimes it's `city`, plus you can't rely on the currency to accurately filter because not all European countries use the Euro, etc. In the US, `state` is very important but currency is all USD. for NZ, the `NZD` currency functionally is the same as an NZ location filter so it seems like overkill, but we don't want to design for that as it's not the case in many places. I think `city` is universally important and we should definitely design for it.

I'm having trouble making universal comments about tags/filters because the needs are quite different depending on what you're searching for. If we're thinking about searching Collectives, we wants things like location and topic areas (e.g. `mutual aid`), but if we're talking about Hosts, legal entity structure is very important, and that doesn't apply to Collectives at all. 

For Hosts we're going to run into tension between accuracy and accessibility, e.g. `non-profit` is not the same thing as `charity` legally (OSC and OCNZ are nonprofits, but not charities), and this is very important for things like accessing certain grants (a big reason people go searching for a fiscal host), but searchers may not understand this distinction and just use `non-profit`.

**Comment by Memo-Es** [comment]
@alanna those last things you mention are tags in my opinion, that way different hosts can either tag`non-profit` or `charity`, same case for `mutual aid` for Collectives. I don't think we need to create additional filters for that

**Comment by kaylarep** [comment]
@SudharakaP @Memo-Es [here's ](https://docs.google.com/spreadsheets/d/1xwe14AFWKZ4wCpInhQBTsZtlDy6A5_KW8J07QbgnIN4/edit?usp=sharing )my working doc for tagging collectives (apologies its a mess): 
- so yes, 'community' is pretty much all of them. I'm trying to tag US regions thoughtfully, like 'Midwest USA' instead of 'Midwest' because 'midwest' is obviously too vague for our global platform.  this is also where i think the tagging cloud will be helpful, so hopefully users can see the 'midwest usa' has been used instead of 'midwest'
- i initially was using [this doc](https://docs.google.com/document/d/1YDs5N7XOFnVD0SYN92X7vMnWIwXbFlfCA84ldYE2ICI/edit?usp=sharing) to select what words to use for the tags, but yes, it is more nuanced than this, and I wouldn't want to label a group with a tag they wouldn't use for themselves. so my strategy now is to just surface words they've used in their 'About' section (since the About section text isn't included in the search results) or off their own websites. 
- +1 to sort by Budget (or maybe by total funds raised?) - both Ascending and Descending (simple arrows are fine I think)
   - would it also be interesting helpful to sort by: amount disbursed? (as a user, i might be curious about this. this would tell me which groups are using the money they're raising, might make me more inclined to donate/contribute to them) 
- @Memo-Es do we have a working list of our current plans for which Filters we will use? I'm a bit lost

**Comment by Memo-Es** [comment]
@kaylarep, here are some thoughts, let me know if it makes sense:
* If `community` is applied for every Collective, then it's not doing anything useful
* Regions can indeed be tags, at least at first before going too deep in the granularity of the location filter, that way we can have tags such as `midwest-USA` `BC-Canada` or anything similar, Collectives need to set this manually
* Users could only see `midwest-USA` if they search for it, I don't see why that should be front-facing everyone
* We would also need a campaign to encourage existing Collectives to add tags, the task for us to add tags to Collectives is out of the question
* Budget or total funds raised sounds good as a filter for Collectives
* I don't understand what you are talking about with 
> @Memo-Es do we have a working list of our current plans for which Filters we will use? I'm a bit lost

**TL;DR: we need a filtering system as simple as possible, only location and tags are enough IMO**

**Comment by SudharakaP** [comment]
> TL;DR: we need a filtering system as simple as possible, only location and tags are enough IMO

From these discussions, I think for the MVP we could do the following, 

1) We need a way to filter by a region (may it be country, region, state or city). 

Here my thinking is the simplest to implement for the MVP is the country filter as collectives already have the ability to insert the country, but I think we need to surface adding this location more, maybe in the collective creation steps. 

Issue: https://github.com/opencollective/opencollective/issues/5332

2) Surface the most used tags and display them in the search page 

For these two I am proposing we use the design proposed by memo on last Friday; 

![image](https://user-images.githubusercontent.com/12435965/158693504-2a714c50-2507-451d-8341-89edfbac9d70.png)

Issue: https://github.com/opencollective/opencollective/issues/5333

**Comment by kaylarep** [comment]
@Memo-Es thanks!
> @kaylarep, here are some thoughts, let me know if it makes sense:
> 
> * If `community` is applied for every Collective, then it's not doing anything useful

**agreed**.

> * Regions can indeed be tags, at least at first before going too deep in the granularity of the location filter, that way we can have tags such as `midwest-USA` `BC-Canada` or anything similar, Collectives need to set this manually

**yes, sorry for lack of clarity. things like 'midwest-USA' would only be a tag, not in any drop-down.**

> * Users could only see `midwest-USA` if they search for it, I don't see why that should be front-facing everyone

**in my mind, midwest-USA would only appear in the tag cloud (if its frequently used enough), or it would appear as a suggestion when you're adding tags.  so if someone wants to search 'midwest', but 'midwest-USA' pops up, they will know to just select that one.  does this make sense?**

> * We would also need a campaign to encourage existing Collectives to add tags, the task for us to add tags to Collectives is out of the question

**yes for sure. OCF decided we will manually add tags to our currently hosted collectives to get the ball rolling, we're just doing this as a one-time project**

**Comment by kaylarep** [comment]
@Memo-Es 
> * I don't understand what you are talking about with
> 
> > @Memo-Es do we have a working list of our current plans for which Filters we will use? I'm a bit lost
> 
> **TL;DR: we need a filtering system as simple as possible, only location and tags are enough IMO**
**yes I agree. I was getting a bit lost, but yes think with our categories on top (fiscal host, collective etc.); location drop down, and tags- that will be enough**

**Comment by kaylarep** [comment]
@SudharakaP :
> Here my thinking is the simplest to implement for the MVP is the country filter as collectives already have the ability to insert the country, but I think we need to surface adding this location more, maybe in the collective creation steps.

+1 to having location in collective creation steps (also considering some groups don't have a location/virtual/digital/remote/whatever the word is)

**Comment by kaylarep** [comment]
another use case in the 'seeking host' category:
> Our group is international, so we also need the ability to send funds outside the U.S. This looks like something the [@oc_foundation](https://twitter.com/oc_foundation) could not support, but we'll keep looking for other options on the [@opencollect](https://twitter.com/opencollect) platform!

**Comment by alinamanko** [comment]
@SudharakaP there is something odd happening which I think is impacted by the work done now on the search functionality. When I am trying to add funds to a collective (for contributions received through bank transfer/check) and need to create a new profile (be it individual or organization), I have a problem to find the country when I am putting in the address. It seems like the countries are not properly coming up. Here is an example when I am trying to input United States: 
![screenshot](https://user-images.githubusercontent.com/57156267/169160133-b0b5ca40-40a5-4a27-bd57-9d0d0ba3f469.png)

Yesterday I had an option to scroll down and countries were listed in alphabetical order, but today they seem to be arranged quite sporadically. And no matter what variation of the name I tried to put down, I couldn't lend on the right option and results coming up don't make sense. Can you please have a look?

**Comment by SudharakaP** [comment]
@alinamanko : Thanks for reporting this issue. The search works by country name. So if you are entering say `us` that won't work as the country name is `United States of America`. If you type say type `Un` then we see all the countries which has `Un` in their name somewhere. And those results are arrange in alphabetical order. The point is that `US` being the country code worked before as we were displaying country codes in the list. But now we don't display the country codes and thus searching by country code isn't supported. Let me know if this is an issue, we can discuss and maybe change the behavior if the consensus here is to also search by country codes like `US`. 🤔

**Comment by alinamanko** [comment]
Thanks @SudharakaP!! I think this is the option I was using before until it changed to country codes recently. I think I also tried typing the name in the old format before and it didn't bring up the results I wanted but trying now it seems to work.

**Comment by SudharakaP** [comment]
The feedback mentioned in this issue has been handled and thus I am closing this issue. If there's any further feedback please feel free to open follow up issues or post at, https://opencollective.slack.com/archives/C035S573ZD2
